### ISSUE TITLE
Vuser Not fully started

### ISSUE THREAD

**Issue by WANGJIANGA** [naver/ngrinder#528]
When vuser Settings are high, they cannot all be started

**Comment by WANGJIANGA** [comment]
The Agent's CPU and Mem are normal

**Comment by WANGJIANGA** [comment]
![image](https://user-images.githubusercontent.com/36441224/77515916-61ab3400-6eb4-11ea-8b50-fbc49a00e33c.png)

**Comment by songeunwoo** [comment]
Hi.
If the agent's open-file is not captured enough, thread execution will fail.
Please check the ulimit of the agent.
https://unix.stackexchange.com/questions/230346/how-to-check-ulimit-usage/410148

and 
I think your Agent environment is 4Core / 4G RAM, 10,000 is an appropriate value.
Please use it appropriately for your agent environment.

**Comment by WANGJIANGA** [comment]
> Hi.
> If the agent's open-file is not captured enough, thread execution will fail.
> Please check the `ulimit` of the agent.
> https://unix.stackexchange.com/questions/230346/how-to-check-ulimit-usage/410148
> 
> and
> I think your Agent environment is 4Core / 4G RAM, 10,000 is an appropriate value.
> Please use it appropriately for your agent environment.

I used 100 8-core 16G agents, process * threads =20*10

**Comment by WANGJIANGA** [comment]
> Hi.
> If the agent's open-file is not captured enough, thread execution will fail.
> Please check the `ulimit` of the agent.
> https://unix.stackexchange.com/questions/230346/how-to-check-ulimit-usage/410148
> 
> and
> I think your Agent environment is 4Core / 4G RAM, 10,000 is an appropriate value.
> Please use it appropriately for your agent environment.

There's no problem with the open-file，Are there other reasons for this？

**Comment by WANGJIANGA** [comment]
> Hi.
> If the agent's open-file is not captured enough, thread execution will fail.
> Please check the `ulimit` of the agent.
> https://unix.stackexchange.com/questions/230346/how-to-check-ulimit-usage/410148
> 
> and
> I think your Agent environment is 4Core / 4G RAM, 10,000 is an appropriate value.
> Please use it appropriately for your agent environment.

Process * thread starts normally after changing to 10*20, what is the reason?

**Comment by songeunwoo** [comment]
> There's no problem with the open-file，Are there other reasons for this？

This is often the case if you don't modify the maximum number of open files.
So I told you to check the ulimit.

It is recommended to use the agent machine for testing purposes only.
Please make sure that no other processes are floating.

**Comment by WANGJIANGA** [comment]
明白，但是我检查了，使用的虚拟机 agent的open-file 设置是足够大的

**Comment by WANGJIANGA** [comment]
> > There's no problem with the open-file，Are there other reasons for this？
> 
> This is often the case if you don't modify the maximum number of open files.
> So I told you to check the ulimit.
> 
> It is recommended to use the agent machine for testing purposes only.
> Please make sure that no other processes are floating.

Yes, but I checked that the open-file Settings of the virtual machine agent used are large enough

**Comment by WANGJIANGA** [comment]
> > There's no problem with the open-file，Are there other reasons for this？
> 
> This is often the case if you don't modify the maximum number of open files.
> So I told you to check the ulimit.
> 
> It is recommended to use the agent machine for testing purposes only.
> Please make sure that no other processes are floating.

I'm sure no other process is floating

**Comment by songeunwoo** [comment]
What version of ngrinder are you running?
I will test it too.

**Comment by WANGJIANGA** [comment]
> 你在运行什么版本的磨床？
> 我也会测试的。

3.4.1

> What version of ngrinder are you running?
> I will test it too.

3.4.1

**Comment by WANGJIANGA** [comment]
> What version of ngrinder are you running?
> I will test it too.

3.4.1

**Comment by songeunwoo** [comment]
I tried with 3.4.4 and there was no problem.
Why don't you try it with 3.4.4?
https://github.com/naver/ngrinder/releases/tag/ngrinder-3.4.4-20200130

**Comment by WANGJIANGA** [comment]
> I tried with 3.4.4 and there was no problem.
> Why don't you try it with 3.4.4?
> https://github.com/naver/ngrinder/releases/tag/ngrinder-3.4.4-20200130

We use 3.4.1
